[D2] CountdownCardView 구현

kimkyunghun3 opened the pull request:
# 개요
- 🔗  이슈링크 : #5 


# 변경사항

## 작업 내용
- CountdonwCardView 구현 ( 사용처에 맞게 로직 변경은 내일 모든 곳에서 함께 적용하겠습니다 )
- timeLabel에서 숫자마다 사이즈가 달라서 가장 긴 숫자의 width를 고정으로 주어서 해결했습니다.

### 스크린샷
| border 적용 시 | border 없을 시|
|:---:|:---:|
|<img src ="https://user-images.githubusercontent.com/52434820/181885166-e27724cd-d2cf-499f-a0a4-e1efb593f3aa.png" width ="200">|<img src ="https://user-images.githubusercontent.com/52434820/181885322-90dfc3e0-6023-464b-817f-b7d43a82a270.png" width ="200">|

kimkyunghun3 committed:
:bulb: CountdownCardViewStyle 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: CountdownCardView 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/DesignSystem/Sources/Component/CountdownCard/DDipCountdownCardViewStyle.swift:
```diff
@@ -0,0 +1,19 @@
+import UIKit
+
+public struct DDipCountdownCardViewStyle {
+    public let iconImage: String
```
이미지는 string이 아닌 DDIPAsset.name을 받으면 좋을것 같아요!

이미지는 어차피 DesignSystem의 Asset으로만 사용하니까요!
(enum인데 대문자로 시작하지 않네요.. 나중에 고쳐놓겠습니다 😭)

kimkyunghun3 replied:
넵넵! 타입 변경해서 반영했습니다!!!

szzang-kr commented on Projects/DesignSystem/Sources/Component/CountdownCard/DDipCountdownCardView.swift:
```diff
@@ -0,0 +1,190 @@
+    
+    public let entireTimeStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.alignment = .fill
```
마찬가지로 지정하지 않아도 될것같습니다.

kimkyunghun3 replied:
넵 수정했습니다!

szzang-kr commented on Projects/DesignSystem/Sources/Component/CountdownCard/DDipCountdownCardView.swift:
```diff
@@ -0,0 +1,190 @@
+    
+    public let secondStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.alignment = .fill
+        stackView.distribution = .fill
```
alignment와 distrubution의 .fill은 기본속성이므로 따로 지정하지 않아도 될것같습니다.

https://developer.apple.com/documentation/uikit/uistackview/1616243-alignment

https://developer.apple.com/documentation/uikit/uistackview/1616233-distribution

kimkyunghun3 replied:
아하! 맞네요!! 제거하겠습니다!

szzang-kr reviewed (changes requested):
수고하셨습니다 🫡

AhnSangHee commented on Projects/DesignSystem/Sources/Component/CountdownCard/DDipCountdownCardView.swift:
```diff
@@ -0,0 +1,190 @@
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = .designSystem(.pretendard, family: .bold, size: ._20)
+        label.textColor = .designSystem(.neutralBlack)
+        label.text = "응모 마감까지"
```
❓ 이 라벨은 내용이 바뀔 일이 없지 않나요?! `public`으로 하신 이유가 궁금합니다!

kimkyunghun3 replied:
네 바뀔일이 없네요! private으로 하는게 더 적절한거같습니다 바꿨습니다~

AhnSangHee reviewed (approved):
수고하셨습니다👍

kimkyunghun3 committed:
:recycle: stackView alignment, distribution 기본제공값 제거

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: titleLabel 접근제어자 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: imageView asset 타입으로 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성
